src/update_flash.c

```diff
@@ -194,6 +194,91 @@ static int RAMFUNCTION wolfBoot_copy_sector(struct wolfBoot_image *src,
     return pos;
 }
 
+static int wolfBoot_swap_and_final_erase(int resume)
+{
+    struct wolfBoot_image boot[1];
+    struct wolfBoot_image update[1];
+    struct wolfBoot_image swap[1];
+    uint8_t st;
+    int eraseLen = WOLFBOOT_SECTOR_SIZE
+#ifdef NVM_FLASH_WRITEONCE
+    /* need to erase the redundant sector too */
+        * 2
+#endif
+    ;
+    int swapDone = 0;
+    uintptr_t tmpBootPos = WOLFBOOT_PARTITION_SIZE - eraseLen -
+        WOLFBOOT_SECTOR_SIZE;
+    /* final swap and erase flag is WOLFBOOT_MAGIC_TRAIL */
+    uint8_t tmpBuffer[sizeof(WOLFBOOT_MAGIC_TRAIL)
+#ifdef EXT_ENCRYPTED
+        + ENCRYPT_KEY_SIZE + ENCRYPT_NONCE_SIZE
+#endif
+    ];
+    /* open boot */
+    wolfBoot_open_image(boot, PART_BOOT);
+    /* open update */
+    wolfBoot_open_image(update, PART_UPDATE);
+    /* open swap */
+    wolfBoot_open_image(swap, PART_SWAP);
```

Comment:
ifndef NO_BACKUP before accessing SWAP

Reply:
wolfBoot_swap_and_final_erase is not called when DISABLE_BACKUP is enabled since it doesn't have the same FSM that requires special handling of the final state setting